src/NServiceBus.Core/Features/Feature.cs

```diff
@@ -64,7 +64,7 @@ protected void Defaults(Action<SettingsHolder> settings)
         /// <summary>
         ///     Called when the features is activated.
         /// </summary>
-        protected internal abstract void Setup(FeatureConfigurationContext context);
+        protected internal abstract IReadOnlyCollection<FeatureStartupTask> Setup(FeatureConfigurationContext context);
```

Comment:
What is the benefit of having the setup return the list of tasks instead of having a "Add" method on the `FeatureConfigurationContext ` ?

Ie: `context.RegisterStartupTask(new MyStartupTask(myDep))`